modules/js/open-close.js

```diff
@@ -1,535 +1,560 @@
-}
+  const bottom = compositeArray[0].constraints[0].pointA.y - rectBase;
+  const top = compositeArray[0].bodies[0].position.y - 200 - pivotValue;
+  const b = top - bottom;
+  const a = compositeArray[2].width;
+  const angleC = Math.acos((a * a + b * b - c * c) / (2 * a * b));
+  if (angleC) {
+    if (angleC - 1.5708 < -1) {
+    } else if (
+      compositeArray[2].bodies[0].angularVelocity > 0.1 ||
+      compositeArray[2].angularVelocity < -0.1
+    ) {
+      Body.setAngularVelocity(compositeArray[2].bodies[0], 0);
+      Body.setAngularVelocity(compositeArray[3].bodies[0], 0);
+    } else {
+      Body.setAngle(compositeArray[2].bodies[0], angleC - 1.5708);
+      Body.setAngle(compositeArray[3].bodies[0], -(angleC - 1.5708));
+    }
+  }
+  Body.setVelocity(compositeArray[2].bodies[0], { x: 0, y: 0 });
+  Body.setVelocity(compositeArray[3].bodies[0], { x: 0, y: 0 });
+  if (compositeArray[1].alternate == false && rackPinionMod == true) {
+    if (compositeArray[1].bodies[0].angle >= 2 * Math.PI) {
+      Body.setAngle(compositeArray[1].bodies[0], 0);
+    }
+    if (
+      compositeArray[1].bodies[0].angle > Math.PI + 0.5 ||
+      compositeArray[1].bodies[0].angle < 0.5
+    ) {
+      if (
+        compositeArray[0].bodies[0].position.y >=
+        compositeArray[0].constraints[0].pointA.y
+      ) {
+        Body.setPosition(compositeArray[0].bodies[0], {
+          x: compositeArray[0].bodies[0].position.x,
+          y: compositeArray[0].constraints[0].pointA.y,
+        });
+        Body.setVelocity(compositeArray[0].bodies[0], { x: 0, y: 0 });
+      } else {
+        Body.setVelocity(compositeArray[0].bodies[0], { x: 0, y: 3 });
+      }
+    }
+  }
+});
 
 // Events.on(engine, 'beforeUpdate', function(event){
 //   yDistance()
 // })
 ////////////////////// RUN /////////////////////////////
 
 // add initial bodies when first loads
-addLinGearComposite((window.innerWidth)*(0.75*0.45),(window.innerHeight)*(0.8) + rackPinBase)
+addLinGearComposite(
+  window.innerWidth * (0.75 * 0.45),
+  window.innerHeight * 0.8 + rackPinBase
+);
 // set constraint stiffness very low
 compositeArray[0].constraints[0].stiffness = 0.0000001;
 // create ui constraints between ui bodies
-createUIConstraints(compositeArray[0], prevSpaceValue, prevPivotValue,6)
+createUIConstraints(compositeArray[0], prevSpaceValue, prevPivotValue, 6);
 // add gear body
-addGearComposite((window.innerWidth)*(0.75*0.45)+((radius)+((toothHeight)*2)) ,(window.innerHeight)*(0.68) + rackPinBase)
+addGearComposite(
+  window.innerWidth * (0.75 * 0.45) + (radius + toothHeight * 2),
+  window.innerHeight * 0.68 + rackPinBase
+);
 // add rectangle bodies
-addRectComposite((300), 5,(window.innerWidth)*(0.75*0.45)-200,compositeArray[0].constraints[0].pointA.y-rectBase)
-addRectComposite((-300), 5,(window.innerWidth)*(0.75*0.45)+200,compositeArray[0].constraints[0].pointA.y-rectBase)
+addRectComposite(
+  300,
+  5,
+  window.innerWidth * (0.75 * 0.45) - 200,
```

Comment:
Magic numbers everywhere!  This is one of the reasons this code will be hard to change.  It is unclear what any of these magic number constants mean.